### PR TITLE
LibWeb: Add support for 'arraybuffer' message types on WebSockets

### PR DESCRIPTION
**Pull request by ben-abraham**
Support for websockets set to 'arraybuffer' mode, receiving non-text data. This is used by discord for communication with the backend.

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/WebSocket.cpp [hunk 1]
```diff
@@ -215,8 +216,21 @@ void WebSocket::on_message(ByteBuffer message, bool is_text)
...
     }
-    // type indicates that the data is Binary and binaryType is "blob"
-    // type indicates that the data is Binary and binaryType is "arraybuffer"
+    
+    if(m_binary_type == "blob") {
+        // type indicates that the data is Binary and binaryType is "blob"
+        TODO();
+    }else if(m_binary_type == "arraybuffer") {
+        // type indicates that the data is Binary and binaryType is "arraybuffer"
+        MessageEventInit event_init {};
```
Comment (alimpfard) [inline Userland/Libraries/LibWeb/HTML/WebSocket.cpp:225]: This looks a bit silly, how about just
```c++
MessageEventInit event_init {
    .data = JS::ArrayBuffer::...,
    .origin = url()
};
```

**Review by alimpfard — COMMENTED** [review]
Pls run clang-format :P

**Comment by alimpfard** [comment]
Also, the commit message could perhaps be something like "LibWeb: Add support for the 'arraybuffer' websocket message type"?

File: Userland/Libraries/LibWeb/HTML/WebSocket.cpp (hunk 1 above)
Comment (ben-abraham) [reply Userland/Libraries/LibWeb/HTML/WebSocket.cpp]: Good idea, I changed the one I templated off of a few lines above as well, just to keep things consistent.

File: Userland/Libraries/LibWeb/HTML/WebSocket.cpp [hunk 2]
```diff
@@ -209,14 +210,30 @@ void WebSocket::on_message(ByteBuffer message, bool is_text)
...
     }
-    // type indicates that the data is Binary and binaryType is "blob"
-    // type indicates that the data is Binary and binaryType is "arraybuffer"
+
+    if (m_binary_type == "blob") {
+        // type indicates that the data is Binary and binaryType is "blob"
+        TODO();
+    } else if (m_binary_type == "arraybuffer") {
+        // type indicates that the data is Binary and binaryType is "arraybuffer"
+        auto& global_obj = wrapper()->global_object();
```
Comment (IdanHo) [inline Userland/Libraries/LibWeb/HTML/WebSocket.cpp:226]: [Please do not use abbreviations](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#wrong). This should be `global_object`.

File: Userland/Libraries/LibWeb/HTML/WebSocket.cpp [hunk 3]
```diff
@@ -209,14 +210,30 @@ void WebSocket::on_message(ByteBuffer message, bool is_text)
...
+        MessageEventInit event_init
+        {
```
Comment (IdanHo) [inline Userland/Libraries/LibWeb/HTML/WebSocket.cpp:228]: Please run clang-format on your code.

**Comment by ben-abraham** [comment]
Sorry for all the revisions. Still getting the whole process down.....
Had to switch back to setting individual properties as it wanted me to pre-define all 6 properties up front (3 direct + 3 from parent)

**Pull request merged**